### PR TITLE
feat: ios version implemented

### PR DESCRIPTION
**Pull request by Dekkee**
iOS implementation of `flutter_radio`
Driven by:
https://github.com/dooboolab/flutter_sound
https://github.com/alexelisenko/audio_player_service/
There is a background audio support and lock screen controls

### CODE REVIEW

File: example/lib/main.dart [hunk 1]
```diff
@@ -24,7 +24,8 @@ class _MyAppState extends State<MyApp> {
   }
 
   Future<void> audioStart() async {
-    await FlutterRadio.audioStart();
+    final item = AudioPlayerItem(title: "Flutter Radio Example", duration: Duration(seconds: 0), progress: 0);
+    await FlutterRadio.audioStart(item);
```
Comment (Dekkee) [inline example/lib/main.dart]: it is a mock for lock screen, while stream is loading. Probably this parameter should be optional

File: lib/flutter_radio.dart [hunk 2]
```diff
@@ -1,36 +1,176 @@
...
+
+    String result = await _channel.invokeMethod('stop');
+    _removePlayerCallback();
+    return result;
   }
 
   static Future<bool> isPlaying() async {
-    final bool playing = await _channel.invokeMethod('isPlaying');
-    return playing;
+    return Future.value(_isPlaying);
```
Comment (Dekkee) [inline lib/flutter_radio.dart:79]: tbh Future version of `isPlaying` flag is unusable. You cant observe it and change UI instantly when in changed. I put it for backward compatibility, but also `stateChanged` stream is implemented.

File: lib/flutter_radio.dart [hunk 3]
```diff
@@ -1,36 +1,176 @@
...
+  static Future<void> _removePlayerCallback() async {
+    if (_playerController != null) {
+      _playerController
+        ..add(null)
+        ..close();
+      _playerController = null;
+    }
+  }
+
+  static Future<String> setMeta(AudioPlayerItem item) async {
```
Comment (Dekkee) [inline lib/flutter_radio.dart:108]: In my case stream doesnt have any information of current media info. It is fetched from another service, so i need to update current meta externally. Probably should be implemented on Android version too.

File: lib/flutter_radio.dart [hunk 4]
```diff
@@ -1,36 +1,176 @@
...
+    String result = await _channel.invokeMethod('setMeta', <String, dynamic>{
+        'meta': item.toMap(),
+      });
+    _removePlayerCallback();
+    return result;
+  }
+
+  static Future<String> setVolume(double volume) async {
```
Comment (Dekkee) [inline lib/flutter_radio.dart:116]: on iOS if you start playing with mute selector on, there will be no sound until you unmute your device. It is wrong behavior for media player. Probably should be implemented in Android version too, or I'll move it inside Native code.

**Comment by Dekkee** [comment]
In my app I split Android and iOS implementations, because I need to implemented background audio and lock screen features on both platforms. Currently, `flutter_radio` doesn't support it. I realise it with https://pub.dartlang.org/packages/audio_service abstraction. But it's not applicable for iOS realisation. Probably I will return back with PR for Android support. But I haven't much time to do it right now.

**Review by thyagoluciano — APPROVED** [review]

**Pull request merged**